### PR TITLE
Use GA Google provider for Terraform sample

### PR DESCRIPTION
**Pull request by jbrook**
### Background 
Updated the samples to use the GA Google provider and to rely on the provider's default Google Cloud project.

### Change Summary
* Use the GA Google provider - all of the resources used are now GA.
* Rely on the provider's default project to avoid setting a global project_id variable.
* Updated instructions to set GOOGLE_CLOUD_PROJECT environment variable.
* Removed unused region variable.
* Less restrictive version constraint for provider.
* Update instructions to enable Cloud Resource Manager API - it's needed by google_project datasource.

### CODE REVIEW

File: docs/asm-gke-terraform/README.md [hunk 1]
```diff
@@ -14,39 +14,40 @@ The following services will be required for this:
...
-### 3.  Create a `variables.tfvars` to set your project ID for Terraform
-Run the following command to create a `variables.tfvars` file
+
+### 3. Enable the Cloud Resource Manager API
+
 ```
-echo "project_id = \"$PROJECT_ID\"" > terraform.tfvars
+gcloud services enable cloudresourcemanager.googleapis.com
 ```
+The google_project Terraform datasource uses this API to read project metadata.
```
Comment (NimJay) [inline docs/asm-gke-terraform/README.md:36]: **Praise:** Thanks for mentioning this tip about `cloudresourcemanager.googleapis.com`.

File: docs/asm-gke-terraform/main.tf [hunk 2]
```diff
@@ -29,24 +28,22 @@ resource "google_container_cluster" "cluster" {
     google_project_service.project
   ]
 }
-data "google_project" "project" {
-  project_id = var.project_id
-}
+
+data "google_project" "project" {}
```
Comment (NimJay) [inline docs/asm-gke-terraform/main.tf:32]: **Question:** Do we still need this?
```suggestion
```
Feel free to ignore. This is a nit.

**Review by NimJay — APPROVED** [review]
@jbrook, thanks so much for these improvements!
(And sorry for the late review.)

Everything looks good.
All my comments are nits (or praise), so feel free to "resolve" my comments without accepting the suggestion(s) or without responding.

I have not tested `terraform apply` (but I trust that you have). Plus, the `terraform validate` GitHub check passes.
Also, it looks like no tutorials are currently using these samples ([see Google-internal code search](https://source.corp.google.com/search?q=%22asm-gke-terraform%22&sq=google3%2Fgoogledata%2Fdevsite%2Fsite-cloud%2Fen)), so merging should be safe.

File: docs/asm-gke-terraform/main.tf (hunk 2 above)
Comment (jbrook) [reply docs/asm-gke-terraform/main.tf:32]: It's needed to add the project number to the mesh_id label on the cluster. I hope that one day soon the label won't be needed anymore.

**Pull request merged**